sklearn/utils/_array_api.py

```diff
@@ -669,6 +669,41 @@ def _average(a, axis=None, weights=None, normalize=True, xp=None):
     return sum_ / scale
 
 
+def _median(x, axis=None, keepdims=False, xp=None):
+    xp, _ = get_namespace(x, xp=xp)
+    if hasattr(xp, "median"):
+        # `torch.median` takes the lower of the two medians when `x` has even number
+        # of elements, thus we use `torch.quantile(q=0.5)`, which gives mean of the two
+        if _is_xp_namespace(xp, "torch"):
```

Comment:
MIght be worth putting this into a helper function like we have is numpy namespace?